File: app/src/main/java/sample/crypto/bux/com/crypto/di/modules/SourceModule.java
package sample.crypto.bux.com.crypto.di.modules;

import com.bux.crypto.RxCrypto;
import com.bux.crypto.RxCryptoDataManager;
import com.bux.crypto.RxCryptoWebSocketConnectionManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SourceModule {

    @Provides
    @Singleton
    static RxCrypto rxCrypto() {
        return RxCrypto.getInstance();
    }

    @Provides
    @Singleton
    static RxCryptoDataManager rxCryptoDataManager(final RxCrypto rxCrypto) {
        return rxCrypto.getDataManager();
    }

    @Provides
    @Singleton
    static RxCryptoWebSocketConnectionManager rxCryptoWebSocketConnectionManager(
            final RxCrypto rxCrypto) {
        return rxCrypto.getWebSocketConnectionManager();
    }

}

File: app/src/main/java/sample/crypto/bux/com/crypto/di/modules/SourceModule.kt
package sample.crypto.bux.com.crypto.di.modules

import com.bux.crypto.RxCrypto
import com.bux.crypto.RxCryptoDataManager
import com.bux.crypto.RxCryptoWebSocketConnectionManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SourceModule {

    @Provides
    @Singleton
    fun rxCrypto(): RxCrypto {
        return RxCrypto.getInstance()
    }

    @Provides
    @Singleton
    fun rxCryptoDataManager(rxCrypto: RxCrypto): RxCryptoDataManager {
        return rxCrypto.dataManager
    }

    @Provides
    @Singleton
    fun rxCryptoWebSocketConnectionManager(
            rxCrypto: RxCrypto): RxCryptoWebSocketConnectionManager {
        return rxCrypto.webSocketConnectionManager
    }

}